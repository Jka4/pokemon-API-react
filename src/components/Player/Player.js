import React, { useState, useEffect } from 'react';

const sound = new Audio("./sound/pokemon_opening.mp3");

const Player = () => {
  const [playing, setPlaying] = useState(false);

  useEffect(() => {
    !playing ? sound.pause() : sound.play();
  });

  return (
    <div className='player'>
      <button className='playButton' onClick={() => setPlaying(!playing)}>
        {playing ? "PAUSE" : "PLAY"}
      </button>
    </div >
  );
}

export default Player;