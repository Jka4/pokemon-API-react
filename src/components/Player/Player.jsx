import React from "react";
import useSound from "use-sound";
import { Button } from "@material-ui/core";
import PlayCircleFilledIcon from "@material-ui/icons/PlayCircleFilled";
import PauseCircleFilledIcon from "@material-ui/icons/PauseCircleFilled";

import "./styles/style.scss";

const soundUrl =
  "https://raw.githubusercontent.com/Jka4/pokemon-API-react/master/public/sound/pokemon_opening.mp3";

const Player = () => {
  const [play, { stop, isPlaying }] = useSound(soundUrl);

  const playPause = () => {
    isPlaying ? stop() : play();
  };

  return (
    <div className="player">
      <Button
        onClick={playPause}
        className="playButton"
        variant="contained"
        color="secondary"
      >
        {isPlaying ? <PauseCircleFilledIcon /> : <PlayCircleFilledIcon />}
      </Button>
    </div>
  );
};

export default Player;
