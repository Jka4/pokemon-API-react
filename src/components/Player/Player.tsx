import React from "react";
import useSound from "use-sound";
import { Button } from "@material-ui/core";
import PlayCircleFilledIcon from "@material-ui/icons/PlayCircleFilled";
import PauseCircleFilledIcon from "@material-ui/icons/PauseCircleFilled";
import Tooltip from "@material-ui/core/Tooltip";

import "./styles/style.scss";

const Player: React.FC = () => {
  const [play, { stop, isPlaying }] = useSound(
    `${process.env.PUBLIC_URL}/sound/pokemon_opening.mp3`
  );

  const playPause = () => {
    isPlaying ? stop() : play();
  };

  return (
    <div className="player">
      <Tooltip
        title="Pokemon opening song (rus)"
        aria-label="Pokemon opening song (rus)"
        enterDelay={500}
        leaveDelay={200}
        arrow
      >
        <Button
          onClick={playPause}
          className="playButton"
          variant="contained"
          color="secondary"
        >
          {isPlaying ? <PauseCircleFilledIcon /> : <PlayCircleFilledIcon />}
        </Button>
      </Tooltip>
    </div>
  );
};

export default Player;
