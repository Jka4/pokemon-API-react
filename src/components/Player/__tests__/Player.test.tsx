import * as React from 'react';
import { shallow } from 'enzyme';
import Player from '../Player';

describe('<Player />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Player />);
    expect(wrapper).toMatchSnapshot();
  });
});
