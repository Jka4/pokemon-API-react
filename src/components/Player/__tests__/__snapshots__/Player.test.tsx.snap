// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders correctly 1`] = `
<div
  className="playerWrapper"
>
  <WithStyles(ForwardRef(Button))
    color="secondary"
    onClick={[Function]}
    variant="contained"
  >
    <Memo(ForwardRef(PlayCircleFilledIcon)) />
  </WithStyles(ForwardRef(Button))>
</div>
`;
